File: Module 07/Lab/StarWars_Common/src/main/java/edu/neumont/interfaces/IApi.java
package edu.neumont.interfaces;

import edu.neumont.dtos.DtoBase;

import java.util.List;

public interface IApi<T extends DtoBase>
{
    List<T> getAll(String url);
    List<T> getPage(String url);
    List<T> get(String url);
}

File: Module 07/dotNet/StarWars.Common/StarWars.Common/Interfaces/IApi.cs
using System;
using System.Collections.Generic;
using StarWars.Common.Dtos;

namespace StarWars.Common.interfaces
{
    public interface IApi<T> where T: DtoBase
    {
        List<T> GetAll();
        List<T> GetPage(int pageNumber);
        List<T> GetPage(string url);
        T Get(string url);
        T Get(int id);
    }
}